packages/tic-tac-toe/contracts/tests/tic-tac-toe.test.ts

```diff
@@ -0,0 +1,293 @@
+import {expectRevert} from '@statechannels/devtools';
+import TicTacToeArtifact from '../../build/contracts/TicTacToe.json';
+import * as ethers from 'ethers';
+import {Contract} from 'ethers';
+import {bigNumberify, Interface} from 'ethers/utils';
+import dotEnvExtended from 'dotenv-extended';
+import path from 'path';
+import {AddressZero} from 'ethers/constants';
+import {TransactionRequest} from 'ethers/providers';
+import {
+  Allocation,
+  encodeOutcome,
+  AssetOutcomeShortHand,
+  replaceAddressesAndBigNumberify,
+  randomExternalDestination
+} from '@statechannels/nitro-protocol';
+import {VariablePart} from '@statechannels/nitro-protocol';
+import {TTTData, PositionType, encodeTTTData} from './types/app-data';
+
+import loadJsonFile from 'load-json-file';
+import {JsonValue} from 'type-fest';
+
+dotEnvExtended.load();
+
+jest.setTimeout(20000);
+
+const testProvider = new ethers.providers.JsonRpcProvider(
+  `http://localhost:${process.env.GANACHE_PORT}`
+);
+
+const PositionIndex = {
+  Start: PositionType.Start,
+  XPlaying: PositionType.XPlaying,
+  OPlaying: PositionType.OPlaying,
+  Draw: PositionType.Draw,
+  Victory: PositionType.Victory
+};
+
+let TicTacToe: Contract;
+
+const numParticipants = 2;
+const addresses = {
+  // participants
+  A: randomExternalDestination(),
+  B: randomExternalDestination()
+};
+
+export function setupContracts(
+  provider: ethers.providers.JsonRpcProvider,
+  artifact,
+  address
+): ethers.ethers.Contract {
+  const signer = provider.getSigner(0);
+
+  const contract = new ethers.Contract(address, artifact.abi, signer);
+  return contract;
+}
+
+async function sendTransaction(
+  contractAddress: string,
+  transaction: TransactionRequest
+): Promise<void> {
+  // move to devtools
+  const signer = testProvider.getSigner();
+  const response = await signer.sendTransaction({to: contractAddress, ...transaction});
+  await response.wait();
+}
+
+beforeAll(async () => {
+  TicTacToe = await setupContracts(
+    testProvider,
+    TicTacToeArtifact,
+    process.env.TTT_CONTRACT_ADDRESS
+  );
+});
+
+//      0  |  1  |  2
+//   +-----------------+
+//      3  |  4  |  5
+//   +-----------------+
+//      6  |  7  |  8
+
+// Draw
+//      X0  |  O5  |  X4
+//   +-----------------+
+//      X2  |  O1  |  O7
+//   +-----------------+
+//      O3  |  X6  |  X8
+
```

Comment:
Maybe include some explainer of what this means?

(I assume it shows the entire history of a game in the format  <X,0><moveNo>.